File: src/main/scala/edu/jhu/fcriscu1/als/graphdb/value/EnsemblAlsSnp.scala
package org.nygenome.als.graphdb.value

import org.apache.commons.csv.CSVRecord

case class EnsemblAlsSnp(
                          ensemblGeneId: String, ensemblTranscriptId: String,
                          variantId: String, hugoName: String, distance: Int,
                          alleleVariation: String
                        ) {
}

object EnsemblAlsSnp extends ValueTrait {
  val columnHeadings: Array[String] = Array("Gene stable ID", "Transcript stable ID"
    , "Variant name", "Gene name", "Distance to transcript"
    , "Variant alleles")

  def parseCSVRecord(record: CSVRecord): EnsemblAlsSnp = {
    new EnsemblAlsSnp(
      record.get(columnHeadings(0)),
      record.get(columnHeadings(1)),
      record.get(columnHeadings(2)),
      record.get(columnHeadings(3)),
      record.get(columnHeadings(4)).toInt,
      record.get(columnHeadings(5))
    )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/EnsemblAlsSnp.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

/**
 * Created by fcriscuo on 2/4/20.
 */
data class EnsemblAlsSnp(
        val ensemblGeneId: String, val ensemblTranscriptId: String,
        val variantId: String, val hugoName: String, val distance: Int,
        val alleleVariation: String
) {
    companion object : AlsdbModel {
        private val columnHeadings: List<String> = listOf("Gene stable ID", "Transcript stable ID"
                , "Variant name", "Gene name", "Distance to transcript"
                , "Variant alleles")

        fun parseCSVRecord(record: CSVRecord): EnsemblAlsSnp =
                EnsemblAlsSnp(
                        record.get(columnHeadings[0]),
                        record.get(columnHeadings[1]),
                        record.get(columnHeadings[2]),
                        record.get(columnHeadings[3]),
                        record.get(columnHeadings[4]).toInt(),
                        record.get(columnHeadings[5])
                )

    }

}